tests/runtime/engine/utils.py

```diff
@@ -99,6 +100,14 @@ def run_test(test):
 
             if test.before:
                 before = subprocess.Popen(test.before, shell=True, preexec_fn=os.setsid)
+                if(test.waitpid):
+                    waited=0
+                    with open(devnull, 'w') as dn:
+                        while subprocess.call("pidof %s" % test.waitpid, shell=True, stdout=dn, stderr=dn) != 0:
+                            time.sleep(0.1)
+                            waited+=0.1
+                            if waited > test.timeout:
```

Comment:
don't we get up to 2x test.timeout now? test.Timeout to  spawn this binary and another test.Timeout for bpftrace  to run.

Reply:
If you raise an  timeout  error  instead of breaking  you fix ^^ and also avoid running bpftrace

Reply:
> don't we get up to 2x test.timeout now
yes, I figured going with the existing TIMEOUT value was better than hardcoding. 

> If you raise an timeout error instead of breaking you fix ^^ and also avoid running bpftrace

Smart! I'll make the change.

Reply:
@fbs unless you feel strongly about a separate TIMEOUT for this, i think leaving it is fine. I think in most cases this should find the pid in less than a second, it shouldn't add much time to the CI run and should pretty much never actually hit the TIMEOUT case.